### ISSUE TITLE
default DPI changed in Inkscape 0.92+

### ISSUE THREAD

**Issue by mehtank** [fablabnbg/inkscape-silhouette#49]
[ see: http://wiki.inkscape.org/wiki/index.php/Release_notes/0.92#Important_changes ]

The way units are handled changed in Inkscape 0.92, and it appears to have broken this plugin, but I haven't quite been able to figure out how.  Sometimes things get cut with the correct dimensions, sometimes they're cut too small at 90/96 scale, and sometimes too big at 96/90 scale.  Perhaps there's some assumption on the units that is not valid with the new Inkscape?

**Comment by amittel** [comment]
Hi!
Just using this plugin. I use `mm` as unit and had no problems. But I found a conversion function with a [hardcoded dpi](https://github.com/fablabnbg/inkscape-silhouette/blob/master/sendto_silhouette.py#L135).
If you change it to 96dpi it should work. Just open it in an editor and change it and should work.

**Comment by jnweiger** [comment]
Yes, this dpi change needs to be supported by the plugin. I think the px2mm() function you found is the right place for this. Code needs to be added there, to do effectively the same that we do in 

 https://github.com/fablabnbg/inkscape-paths2openscad/blob/master/paths2openscad.py#L511

A different workaround with Inkscape 0.92 should be 
 -> Extensions -> Document -> DPI 90 to 96

**Comment by lexelby** [comment]
From what I can tell, loading an old file in inkscape 0.92 and selecting "set viewbox" works to correct the dpi change, but *only* if the old document's dimensions were specified in mm or inches.  That is, the `<svg>` tag's `width` and `height` must include units in them.

The reason this works is that this plugin will first interpret those dimensions (incorrectly) in 90 DPI due to `getLength()`, and then interpret the viewbox in pixels, resulting in an identity transform as the `docTransform`.  Then it will interpret all pixel measurements in path objects as if they were specified in 90 DPI, due to `px2mm()`.

In essence, the two places in which the plugin hardcodes 90 DPI cancel each other out!  The viewbox tells inkscape to scale things to work with its 90 DPI, and dimensions are correctly reflected in the UI.  The plugin essentially ignores the viewbox and works in 90 DPI, and that works out too.

However, if the old document's width/height are specified in pixels, then all bets are off.  Inkscape will scale the width/height in the `<svg>` tag by a factor of 1.0666667 (=96/90).  The `docTransform` ends up being `scale(1.0666667, 1.0666667)` and everything is cut too big.  This probably accounts for the inconsistent behavior that I and @mehtank have seen.

**Comment by CoolGames** [comment]
@lexelby ,
I tried View box using the [Reg Marks letter in svg example](https://github.com/nosliwneb/robocut/blob/master/examples/letter_reg-marks.svg) from 2010 version of [Robocut](https://github.com/nosliwneb/robocut).
**They are originally bottom left dot and right corner and top left corner so need to be flipped vertically.**

I created a simple cut outline of a balloon with text located inside the outline.
Registration marks were 180 by 240 mm now in the correct position for successful detection.
I used a pen expecting to trace the printed outline with inkscape-silhouette after turning off reg marks and text layers.
I also edited line 135 of [Send to Silhouette](https://github.com/fablabnbg/inkscape-silhouette/blob/master/sendto_silhouette.py#L135) to be 96 and the plotted balloon was smaller than the printed size and shifted up to right.
_**Did not trace the balloon outlining the text that was printed.**_
Here's the screen shot.
Pen doesn't trace the balloon and the registration marks are physically 180 by 240 +/- 0.5mm.
![export silhouette - screenshot_2017-09-17_19-34-42](https://user-images.githubusercontent.com/149711/30525948-64d99c52-9bdf-11e7-9887-72ff4dfb9b3e.png)


**Simpler test.**
Using [inkscape 0.92](https://gitlab.com/inkscape/inkscape) on Ubuntu Studio 17.04 I created a 4.5 inch circle in another layer on the above example and it was drawn as 4.75 inches with the pen.

**Alignment example**
Uploaded in ZIP file as
[Balloons - RegMarks - TEST Text.zip](https://github.com/fablabnbg/inkscape-silhouette/files/1310603/Balloons.-.RegMarks.-.TEST.Text.zip)

**_RED_** Layer named **outline** duplicated and moved to **_BLACK_** layer  named **REAL cut** then it was expanded by **106.667 percent**.
Using a pen plot successfully traced **_RED_** **outline** layer printer with **Reg Marks** layer using only **_BLACK_** **REAL cut** layer.
.
![balloons - regmarks - test text](https://user-images.githubusercontent.com/149711/30541806-d5e8db7c-9c4a-11e7-9eda-1d173c1ba167.png)

**Comment by EtherGraf** [comment]
Just found the conversion functions self.unittouu() / self.uutounit() from the inkex class
Some hints at 
http://wiki.inkscape.org/wiki/index.php/PythonEffectTutorial
https://inkscape.gitlab.io/inkscape/doxygen-extensions/classinkex_1_1Effect.html

From the description I would expect it to convert the coordinates correctly. 
This function exists from at least 0.48 and should hopefully work with all releases and their dpi-settings.
